src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.reform.blobrouter.services.storage;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.sas.BlobContainerSasPermission;
+import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
+import org.slf4j.Logger;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+
+import static org.slf4j.LoggerFactory.getLogger;
+import static uk.gov.hmcts.reform.blobrouter.services.storage.RejectedFilesHandler.REJECTED_CONTAINER_SUFFIX;
+
+@Component
+public class BlobRejecter {
+
+    private static final Logger logger = getLogger(BlobRejecter.class);
+
+    private final BlobServiceClient storageClient;
+
+    public BlobRejecter(BlobServiceClient storageClient) {
+        this.storageClient = storageClient;
+    }
+
+    public void moveToRejectedContainer(String blobName, String containerName) {
+
+        BlobClient sourceBlob = getBlobClient(containerName, blobName);
+        BlobClient targetBlob = getBlobClient(containerName + REJECTED_CONTAINER_SUFFIX, blobName);
+
+        if (targetBlob.exists()) {
+            targetBlob.createSnapshot();
+        }
+
+        String loggingContext = String.format(
+            "File name: %s. Source Container: %s. Target Container: %s",
+            blobName,
+            sourceBlob.getContainerName(),
+            targetBlob.getContainerName()
+        );
+
+        if (!sourceBlob.exists()) {
+            logger.error("File already deleted. " + loggingContext);
+        } else {
+            copy(targetBlob, sourceBlob, loggingContext);
+            sourceBlob.delete();
+            logger.info("File successfully moved to rejected container. " + loggingContext);
```

Comment:
After `copy` we know that the file has been copied. Maybe in this last `info` we should say that the source file was deleted.

Reply:
moved existing code